spec/lib/active_record_query_trace_spec.rb

```diff
@@ -255,11 +251,39 @@
             User.create!
           end
 
-          it "displays the last #{lines} #{'line'.pluralize(lines)} of the backlog" do
+          it "displays the last #{lines} #{'line'.pluralize(lines)} of the backtrace" do
             expect(log.scan(line).size).to eq(lines)
           end
         end
       end
+
+      it "displays the last 5 lines of the backtrace by default" do
+        User.create!
+
+        expect(log.scan(line).size).to eq(5)
+      end
+
+      it "displays the full backtrace when set to 0" do
+        described_class.lines = 0
+        User.create!
+
+        expect(log.scan(line).size).to eq(backtrace.length)
+      end
+
+      it "displays the full backtrace when log level is set to :full" do
```

Comment:
Again, the log level (`:full`, `:app` or `:rails`) setting is not meant to affect the number of lines that are displayed in the backtrace.